index.ts

```diff
@@ -78,7 +82,7 @@ class Money {
         Object.freeze(this)
     }
 
-    static fromInteger(amount: number|any, currency?: string): Money {
+    static fromInteger(amount: number | Partial<IMoney>, currency?: string): Money {
```

Comment:
I would probably use `IMoney` directly since none of its properties are optional for this function.

```suggestion
    static fromInteger(amount: number | IMoney, currency?: string): Money {
```